Detectors/TPC/reconstruction/include/TPCReconstruction/KrBoxClusterFinder.h

```diff
@@ -0,0 +1,166 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file KrBoxClusterFinder3D.h
+/// \brief Class for Krypton and X-ray events
+/// \author Philip Hauer <hauer@hiskp.uni-bonn.de>
+
+/// Based on old BoxClusterer.cxx which worked in two dimensions
+///
+/// The KrBoxClusterFinder3D basically tries to find local maximums and builds up
+/// clusters around these. So if c is the local maximum (c for center). It
+/// will try to make a cluster like this (shown here only in two dimensions):
+///    --->  pad direction
+///    o o o o o    |
+///    o i i i o    |
+///    o i C i o    V Time direction
+///    o i i i o
+///    o o o o o
+///
+/// The outer pad-time cells are only addded if the inner cell has signal. For
+/// horizonal vertically aligned inner cells we test like this:
+///        o
+///        i
+///    o i C i o
+///        i
+///        o
+/// For the diagonal cells we check like this.
+///    o o   o o
+///    o i   i o
+///        C
+///    o i   i o
+///    o o   o o
+///
+/// The requirements for a local maxima is:
+/// Charge in bin is >= mQThresholdMax ADC channels.
+/// Charge in bin is larger than all digits in the box around it.
+/// (in the case when it is similar only one of the clusters will be made, see
+/// in the code)
+/// At least mMinNumberOfNeighbours neighbours (= direct neighbours) has a signal.
+///
+/// Implementation:
+/// The RAW data is "expanded" for each sector and stored in a big signal
+/// mMapOfAllDigits = 3d vector of all digits
+///
+/// To make sure that one never has to check if one is inside the sector or not
+/// the arrays are larger than a sector. For the size in row-direction, a constant is used.
+/// For time and pad direction, a number is set (here is room for improvements).
+///
+/// When data from a new sector is encountered, the method
+///
+/// How to use:
+/// Load tpcdigits.root
+/// Loop over all events
+/// Loop over all sectors
+/// Use KrBoxClusterFinder(sector) to create 3D map
+/// Use localMaxima() to find..well the local maxima
+/// localMaxima() yields a vector of a tuple with three int
+/// Loop over this vector and give the tuple to buildCluster()
+/// Write to tree or something similar
+
+#ifndef ALICEO2_TPC_KrBoxClusterFinder_H_
+#define ALICEO2_TPC_KrBoxClusterFinder_H_
+
+//#pragma link C++ class std::vector < KrCluster> + ;
+
+#include "TCanvas.h"
+#include "TColor.h"
+#include "TFile.h"
+#include "TH1.h"
+#include "TTree.h"
+
+#include "TPCBase/Digit.h"
+#include "TPCReconstruction/KrCluster.h"
+
+#include <array>
+#include <iostream>

```

Comment:
array and iostream are not used in the header. Please remove. Add in cxx if necessary.

Reply:
Done.